api/src/main/java/org/bahmni/module/hip/api/dao/impl/EncounterDaoImpl.java

```diff
@@ -0,0 +1,164 @@
+    }
+
+    private String sqlGetEncounterIdsForVisitForPrescriptions = "select\n" +
+            "\te.encounter_id\n" +
+            "from\n" +
+            "\tvisit as v\n" +
+            "inner join visit_type as vt on\n" +
+            "\tvt.visit_type_id = v.visit_type_id\n" +
+            "inner join encounter as e on\n" +
+            "\te.visit_id = v.visit_id\n" +
+            "where\n" +
+            "\tvt.name = :visit\n" +
+            "\tand e.encounter_id not in (\n" +
+            "\tselect\n" +
+            "\t\tencounter_id\n" +
+            "\tfrom\n" +
+            "\t\tepisode_encounter)\n" +
+            "\tand v.date_started between :fromDate and :toDate\n" +
+            "\tand v.patient_id = (select person_id from person as p2 where p2.uuid = :patientUUID);";
+
+    private String sqlGetEncounterIdsForProgramForPrescriptions = "select\n" +
+            "\tle.encounter_id\n" +
+            "from\n" +
+            "\tpatient_program as pp\n" +
+            "inner join program as p on\n" +
+            "\tpp.program_id = p.program_id\n" +
+            "inner join (\n" +
+            "\tselect\n" +
+            "\t\tee.episode_id,\n" +
+            "\t\tee.encounter_id,\n" +
+            "\t\tepp.patient_program_id\n" +
+            "\tfrom\n" +
+            "\t\tepisode_encounter as ee\n" +
+            "\tinner join episode_patient_program as epp on\n" +
+            "\t\tee.episode_id = epp.episode_id) as le on\n" +
+            "\tle.patient_program_id = pp.patient_program_id\n" +
+            "where\n" +
+            "\tp.name = :programName \n" +
+            "\tand pp.patient_id = (select person_id from person as p2 where p2.uuid = :patientUUID)\n" +
+            "\tand pp.patient_program_id in (\n" +
+            "\tselect\n" +
+            "\t\tpatient_program_id\n" +
+            "\tfrom\n" +
+            "\t\tprogram_attribute_type as pat\n" +
+            "\tinner join patient_program_attribute as ppa on\n" +
+            "\t\tpat.program_attribute_type_id = ppa.attribute_type_id\n" +
+            "\twhere\n" +
+            "\t\tname = \"ID_Number\"\n" +
+            "\t\tand value_reference = :programEnrollmentId )\n" +
+            "\tand pp.date_enrolled between :fromDate and :toDate ;";
+
+    private String sqlGetEncounterIdsForVisitForDiagnosticReports = "select \n" +
+            "            e.encounter_id \n" +
+            "            from \n" +
+            "            visit as v \n" +
+            "            inner join visit_type as vt on \n" +
+            "            \tvt.visit_type_id = v.visit_type_id \n" +
+            "            inner join encounter as e on \n" +
+            "            \te.visit_id = v.visit_id\n" +
+            "            inner join obs o on \n" +
+            "            \to.encounter_id = e.encounter_id \n" +
+            "            where\n" +
+            "            (e.encounter_type =6 or e.encounter_type =9) \n" +
```

Comment:
Do not hardcode the encounter types & the concept_ids - either fetch from the encouter table or have it as consts.

Reply:
done